### PR TITLE
feat: add icq module & add fee-abs docker tests

### PR DESCRIPTION
**Pull request by atheeshp**
- ~Adds ICQ module, which requires to use fee-abs~, no need to integrate ICQ (it requires on the other network, in this case on osmosis) and no need to open the `icqhost` port on `cheqd`
 - Adds fee-abstraction tests

### CODE REVIEW

File: tests/fee-abs/cheqd/host_zone.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	"description": "Add Fee Abstraction Host Zone",
+	"host_chain_fee_abs_config": 
+		{
+			"ibc_denom": "ibc/ED07A3391A112B175915CD8FAF43A2DA8E4790EDE12566649D0C2F97716B8518",
+			"osmosis_pool_token_denom_in": "uosmo",
+			"pool_id": "1",
+			"status": 0
+		},
+	"deposit": "100000000ncheq"
+}
```
Comment (atheeshp) [inline tests/fee-abs/cheqd/host_zone.json:12]:
```suggestion

}
```

File: tests/fee-abs/cheqd/proposal.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        "value": "channel-1"
+      },
+      {
+        "subspace": "feeabs",
+        "key": "OsmosisCrosschainSwapAddress",
+        "value": "osmo17p9rzwnnfxcjp32un9ug7yhhzgtkhvl9jfksztgw5uh69wac2pgs5yczr8"
+      }
+    ],
+    "deposit": "10000000000ncheq"
+  }
```
Comment (atheeshp) [inline tests/fee-abs/cheqd/proposal.json:32]:
```suggestion

  }
```

File: tests/fee-abs/hermes/set_hermes_accounts.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+hermes keys add --chain osmosis --mnemonic-file ~/.hermes/keys/osmosis/keyring-test/osmo_relayer_mnemonic.txt --key-name osmosis-key --overwrite
+
+info "Open channel"
+
+hermes create channel --a-chain cheqd --b-chain osmosis --a-port transfer --b-port transfer --new-client-connection --yes
+# hermes create connection --a-chain cheqd --b-chain osmosis --new-client-connection --yes
+hermes create channel --a-chain osmosis --b-chain cheqd --a-port icqhost --b-port feeabs --new-client-connection --yes
+# hermes create channel --a-chain cheqd --b-chain osmosis --a-port icqhost --b-port feeabs --new-client-connection --yes
+
+hermes start
```
Comment (atheeshp) [inline tests/fee-abs/hermes/set_hermes_accounts.sh:75]:
```suggestion
hermes start

```

File: tests/fee-abs/osmosis/deploy_osmosis_contract.sh [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+sleep 10
+
+info "instantiate swaprouter"
+# Instantiate the crosschainswap contract
+INIT_CROSSCHAIN_SWAPS='{"swap_contract":"'$SWAPROUTER_ADDRESS'","governor": "'$OWNER'"}'
+RES=$(osmosisd tx wasm instantiate 3 "$INIT_CROSSCHAIN_SWAPS" --keyring-backend=test --home=$HOME/.osmosisd --from osmosis-user --chain-id osmosis --label "test" --no-admin --yes --fees 5000uosmo)
+assert_tx_successful $RES
+sleep 5
+CROSSCHAIN_SWAPS_ADDRESS=osmo17p9rzwnnfxcjp32un9ug7yhhzgtkhvl9jfksztgw5uh69wac2pgs5yczr8
+# 1
```
Comment (atheeshp) [inline tests/fee-abs/osmosis/deploy_osmosis_contract.sh:112]:
```suggestion
# 1

```

File: tests/fee-abs/osmosis/pool.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+    "weights": "10ibc/19D515BB82FEAFCEC357D04C1B75DBF123DB5479041A9BAE38BFCF295477404D,2uosmo",
+    "initial-deposit": "1000ibc/19D515BB82FEAFCEC357D04C1B75DBF123DB5479041A9BAE38BFCF295477404D,200uosmo",
+    "swap-fee": "0.01",
+    "exit-fee": "0",
+    "future-governor": ""
+  }
+
```
Comment (atheeshp) [inline tests/fee-abs/osmosis/pool.json:8]:
```suggestion

```

File: tests/fee-abs/osmosis/send_ibc_and_create_pool.sh [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+info "enable fee abs"
+cheqd-noded tx gov submit-legacy-proposal param-change proposal.json --from $CHEQD_USER_ADDRESS --keyring-backend test --chain-id cheqd --yes --gas-prices 50ncheq --gas 350000
+sleep 5 
+cheqd-noded tx gov vote 1 yes --from $CHEQD_USER_ADDRESS --keyring-backend test --chain-id cheqd --yes --gas-prices 50ncheq
+sleep 5 
+
+info "add host zone config"
+cheqd-noded tx gov submit-legacy-proposal add-hostzone-config host_zone.json --from $CHEQD_USER_ADDRESS --keyring-backend test --chain-id cheqd --yes --gas-prices 50ncheq --gas 350000
+sleep 5
+cheqd-noded tx gov vote 2 yes --from $CHEQD_USER_ADDRESS --keyring-backend test --chain-id cheqd --yes --gas-prices 50ncheq
```
Comment (atheeshp) [inline tests/fee-abs/osmosis/send_ibc_and_create_pool.sh:75]:
```suggestion
cheqd-noded tx gov vote 2 yes --from $CHEQD_USER_ADDRESS --keyring-backend test --chain-id cheqd --yes --gas-prices 50ncheq

```

**Review by atheeshp — COMMENTED** [review]
nits

File: tests/fee-abs/cheqd/proposal.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      {
+        "subspace": "feeabs",
+        "key": "OsmosisCrosschainSwapAddress",
+        "value": "osmo17p9rzwnnfxcjp32un9ug7yhhzgtkhvl9jfksztgw5uh69wac2pgs5yczr8"
+      }
+    ],
+    "deposit": "10000000000ncheq"
+
+  }
```
Comment (atheeshp) [inline tests/fee-abs/cheqd/proposal.json:33]:
```suggestion
  }
  
```

File: tests/fee-abs/osmosis/pool.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+    "weights": "10ibc/19D515BB82FEAFCEC357D04C1B75DBF123DB5479041A9BAE38BFCF295477404D,2uosmo",
+    "initial-deposit": "1000ibc/19D515BB82FEAFCEC357D04C1B75DBF123DB5479041A9BAE38BFCF295477404D,200uosmo",
+    "swap-fee": "0.01",
+    "exit-fee": "0",
+    "future-governor": ""
+  }
```
Comment (atheeshp) [inline tests/fee-abs/osmosis/pool.json:7]:
```suggestion
  }
  
```

**Review by atheeshp — COMMENTED** [review]
nits

**Review by Eengineer1 — APPROVED** [review]
Looks well polished.

Approved.

**Pull request merged**